src/sample_weight_audit/estimator_check.py

```diff
@@ -80,65 +84,46 @@ def check_weighted_repeated_estimator_fit_equivalence(
-    )
-    assert (
-        predictions_weighted.ndim == 3
-    )  # (n_stochastic_fits, n_test_data_points, prediction_dim)
-    assert predictions_weighted.shape == predictions_repeated.shape
-
-    message = (
-        f"Repeatedly fitting {est} with different random seeds led to the "
-        "same predictions: please check sample weight equivalence by an exact "
-        "equality test instead of statistical tests."
+    scores_weighted, scores_repeated, predictions_weighted, predictions_repeated = (
+        multifit_over_weighted_and_repeated(
+            est,
+            n_features=n_features,
+            n_classes=n_classes,
+            n_stochastic_fits=n_stochastic_fits,
+            n_samples_per_cv_group=n_samples_per_cv_group,
+            n_cv_group=n_cv_group,
+            max_sample_weight=max_sample_weight,
+            random_state=random_state,
+        )
     )
-    diffs = predictions_weighted.max(axis=0) - predictions_weighted.min(axis=0)
-    if np.all(diffs < np.finfo(diffs.dtype).eps):
-        raise UnexpectedDeterministicPredictions(message)
-
-    diffs = predictions_repeated.max(axis=0) - predictions_repeated.min(axis=0)
-    if np.all(diffs < np.finfo(diffs.dtype).eps):
-        raise UnexpectedDeterministicPredictions(message)
-
-    assert predictions_weighted.ndim == 3
-    assert predictions_weighted.shape[0] == n_stochastic_fits
-    # assert math.prod(predictions_weighted.shape[1:]) == stat_test_dim
-    assert predictions_repeated.shape == predictions_weighted.shape
-
-    data_to_test_weighted = predictions_weighted.reshape(
-        (n_stochastic_fits, stat_test_dim)
-    ).T  # shape: (stat_test_dim, n_stochastic_fits)
-    data_to_test_repeated = predictions_repeated.reshape(
-        (n_stochastic_fits, stat_test_dim)
-    ).T  # shape: (stat_test_dim, n_stochastic_fits)
 
+    assert scores_weighted.ndim == 1  # (n_stochastic_fits,)
+    assert scores_weighted.shape == scores_repeated.shape
+
+    deterministic_flag = False
+    if predictions_weighted.dtype != np.int32:
+        diffs = predictions_weighted.max(axis=0) - predictions_weighted.min(axis=0)
+        if np.all(diffs < np.finfo(diffs.dtype).eps):
+            deterministic_flag = True
+
+    if predictions_repeated.dtype != np.int32:
+        diffs = predictions_repeated.max(axis=0) - predictions_repeated.min(axis=0)
+        if np.all(diffs < np.finfo(diffs.dtype).eps):
+            deterministic_flag = True
+
+    data_to_test_weighted = scores_weighted
+    data_to_test_repeated = scores_repeated
     # Iterate of all statistical test dimensions and compute p-values
     # for each dimension.
-    pvalues = []
-    for i in range(stat_test_dim):
-        pvalue = run_1d_test(
-            data_to_test_weighted[i], data_to_test_repeated[i], test_name
-        ).pvalue
-        pvalues.append(pvalue)
-
-    pvalues = np.asarray(pvalues)
+    pvalue = run_1d_test(data_to_test_weighted, data_to_test_repeated, test_name).pvalue
```

Comment:
Since we never run ndimensional tests anymore, this function could probably be renamed `run_two_sample_test` or `run_statistical_test`.